contrib/devtools/test-symbol-check.py

```diff
@@ -48,31 +48,31 @@ def test_ELF(self):
                 }
         ''')
 
-        self.assertEqual(call_symbol_check(cc, source, executable, ['-lutil']),
+        self.assertEqual(call_symbol_check(cxx, source, executable, ['-lutil']),
                 (1, executable + ': libutil.so.1 is not in ALLOWED_LIBRARIES!\n' +
                     executable + ': failed LIBRARY_DEPENDENCIES'))
 
         # finally, check a simple conforming binary
-        source = 'test3.c'
+        source = 'test3.cpp'
         executable = 'test3'
         with open(source, 'w', encoding="utf8") as f:
             f.write('''
-                #include <stdio.h>
+                #include <iostream>
```

Comment:
```suggestion
                #include <cstdio>
```

as suggested in https://github.com/bitcoin/bitcoin/pull/30387#issuecomment-2208993921.